step-automation-packages/step-automation-packages-yaml/src/test/java/step/automation/packages/yaml/AutomationPackageDescriptorReaderTest.java

```diff
@@ -46,22 +45,21 @@ public void jmeterKeywordReadTest() throws AutomationPackageReadingException {
         try (InputStream is = new FileInputStream(descriptor)) {
             AutomationPackageDescriptorYaml automationPackage = reader.readAutomationPackageDescriptor(is, "");
             assertNotNull(automationPackage);
-            List<AutomationPackageKeyword> keywords = automationPackage.getKeywords();
+            List<YamlAutomationPackageKeyword> keywords = automationPackage.getKeywords();
             assertEquals(1, keywords.size());
-            AutomationPackageKeyword jmeterKeyword = keywords.get(0);
-            Function k = jmeterKeyword.getDraftKeyword();
-            assertEquals("JMeter keyword from automation package", k.getAttribute("name"));
+            YamlAutomationPackageKeyword jmeterKeyword = keywords.get(0);
+            YamlJMeterFunction k = (YamlJMeterFunction) jmeterKeyword.getYamlKeyword();
+            assertEquals("JMeter keyword from automation package", k.getName());
             assertEquals("JMeter keyword 1", k.getDescription());
             assertFalse(k.isExecuteLocally());
             assertTrue(k.isUseCustomTemplate());
-            assertFalse(k.isManaged());
             assertEquals((Integer) 1000, k.getCallTimeout().get());
             assertNotNull("string", k.getSchema().getJsonObject("properties").getJsonObject("firstName").getJsonString("type"));
 
-            assertEquals("jmeterProject1/jmeterProject1.xml", jmeterKeyword.getSpecialAttributes().get(JMeterFunctionTestplanConversionRule.JMETER_TESTPLAN_ATTR));
+            assertEquals("jmeterProject1/jmeterProject1.xml", k.getJmeterTestplan().get());
 
-            assertEquals("valueA", jmeterKeyword.getDraftKeyword().getTokenSelectionCriteria().get("criteriaA"));
-            assertEquals("valueB", jmeterKeyword.getDraftKeyword().getTokenSelectionCriteria().get("criteriaB"));
+            assertEquals("valueA", k.getRouting().get("criteriaA"));
```

Comment:
@david-stephan 
Yes, it is a good idea to cover this logic with unit tests. But I think we can do this before merging the final solution to master.

It would be nice to prepare the test data (yamls) covering the most of keywords and artefacts and in test try to deserialize this data and store in database.

Reply:
@iegorov777 yes that's a good idea

Reply:
@david-stephan 
The new test for serialization/deserialization of plan with all supported controls has been added to Step OS and Step EE